sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -447,18 +514,65 @@ public String getEventHubName() {
             // Start send span and store updated context
             parentContext.set(tracerProvider.startSpan(finalSharedContext, ProcessKind.SEND));
         }
-
-        return withRetry(getSendLink(batch.getPartitionId())
-            .flatMap(link ->
-                messages.size() == 1
+        if (isIdempotentPartitionPublishing) {
+            PartitionPublishingState publishingState = this.getClientPartitionPublishingState(batch.getPartitionId());
+            return Mono.fromRunnable(() -> {
+                publishingState.getSemaphore().acquireUninterruptibly();
```

Comment:
Does this part need to be in a mono? it seems ysnchronous to me.

Reply:
Yes, without a Mono, it's too eager to have the Semaphore and may cause some problems.